test/e2e/operator_test.go

```diff
@@ -101,6 +122,80 @@ func TestDefaultClusterIngressExists(t *testing.T) {
 	}
 }
 
+func TestMultipleClusterIngressCreate(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Errorf("failed to get DNS 'cluster': %v", err)
+	}
+
+	ing0 := newClusterIngress(clusterIngressName+"0", ns, clusterIngressName+"0"+"."+dnsConfig.Spec.BaseDomain)
+	ing1 := newClusterIngress(clusterIngressName+"1", ns, clusterIngressName+"1"+"."+dnsConfig.Spec.BaseDomain)
+	ingresses := []*ingressv1alpha1.ClusterIngress{ing0, ing1}
+	for _, ing := range ingresses {
+		err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+			if err := cl.Create(context.TODO(), ing); err != nil {
+				if errors.IsAlreadyExists(err) {
+					return true, nil
+				} else {
+					return false, nil
+				}
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Errorf("failed to create ClusterIngress %s/%s: %v", ing.Namespace, ing.Name, err)
+		}
+		err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+			if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}, ing); err != nil {
+				if ing.Status.Replicas != ing.Spec.Replicas {
+					return false, nil
+				} else {
+					return true, nil
+				}
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Errorf("failed to reconcile ClusterIngress %s/%s: %v", ing.Namespace, ing.Name, err)
+		}
+	}
+}
+
+func TestMultipleClusterIngressDelete(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ing0 := &ingressv1alpha1.ClusterIngress{}
+	ing1 := &ingressv1alpha1.ClusterIngress{}
+	ingresses := []*ingressv1alpha1.ClusterIngress{ing0, ing1}
+	for i, ing := range ingresses {
+		name := clusterIngressName + strconv.Itoa(i)
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: name}, ing); err != nil {
+			t.Errorf("failed to get ClusterIngress %s/%s: %v", ns, name, err)
+		}
+		err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+			if err := cl.Delete(context.TODO(), ing); err != nil {
```

Comment:
Are we depending on the test case order here i.e TestMultipleClusterIngressCreate() is called before TestMultipleClusterIngressDelete()? If yes, why not combine these tests as a singe test case?